### PR TITLE
Configure github actions

### PR DESCRIPTION
**Pull request by nullkiller**
We need to replace old Travis-CI with some new CI. Reasons:
- travis-ci.org no longer works
- we constantly exceed Travis limits on build execution time

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -63,7 +63,7 @@ set(PACKAGE_FILE_NAME "" CACHE STRING "Override for CPack package filename")
 #        Miscellaneous options             #
 ############################################
 
-set(CMAKE_MODULE_PATH ${CMAKE_HOME_DIRECTORY}/cmake_modules)
+set(CMAKE_MODULE_PATH ${CMAKE_HOME_DIRECTORY}/cmake_modules ${PROJECT_SOURCE_DIR}/CI)
```
Comment (nullkiller) [inline CMakeLists.txt:66]: For NSIS.template.in

File: lib/mapping/CMap.cpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ SHeroName::SHeroName() : heroId(-1)
 
 PlayerInfo::PlayerInfo(): canHumanPlay(false), canComputerPlay(false),
 	aiTactic(EAiTactic::RANDOM), isFactionRandom(false), hasRandomHero(false), mainCustomHeroPortrait(-1), mainCustomHeroId(-1), hasMainTown(false),
-	generateHeroAtMainTown(false), team(TeamID::NO_TEAM), /* following are unused */ generateHero(false), p7(0), powerPlaceholders(-1)
+	generateHeroAtMainTown(false), posOfMainTown(-1), team(TeamID::NO_TEAM), /* following are unused */ generateHero(false), p7(0), powerPlaceholders(-1)
```
Comment (nullkiller) [inline lib/mapping/CMap.cpp:33]: small fix for unit test. posOfMainTown.valid() should correlate with hasMainTown

File: license.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
-    GNU GENERAL PUBLIC LICENSE
+﻿    GNU GENERAL PUBLIC LICENSE
```
Comment (nullkiller) [inline license.txt:1]: Encoded with UTF-8 BOM for NSIS

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -56,10 +56,6 @@ set(test_SRCS
  		spells/targetConditions/ReceptiveFeatureConditionTest.cpp
  		spells/targetConditions/SpellEffectConditionTest.cpp
  		spells/targetConditions/TargetConditionItemFixture.cpp
-		
-		vcai/mock_ResourceManager.cpp
```
Comment (nullkiller) [inline test/CMakeLists.txt:60]: Need to find a way to test VCAI and other dynamic libs without tons of DLL_LINKAGE

Comment (nullkiller) [reply test/CMakeLists.txt:60]: Anyway tests does not run for now. They require VLC and HOMM3 data

File: test/map/CMapFormatTest.cpp [hunk 5]
```diff
@@ -42,6 +42,12 @@ TEST(MapFormat, Random)
 	SCOPED_TRACE("MapFormat_Random start");
 
 	CMapGenOptions opt;
+	CRmgTemplate tmpl;
+
+	const_cast<CRmgTemplate::CPlayerCountRange &>(tmpl.getCpuPlayers()).addRange(1, 4);
+	const_cast<CRmgTemplate::Zones &>(tmpl.getZones())[0] = std::make_shared<rmg::ZoneOptions>();
```
Comment (nullkiller) [inline test/map/CMapFormatTest.cpp:48]: a few more unit test fixes

File: CI/NSIS.template.in [hunk 6]
```diff
@@ -0,0 +1,973 @@
...
+  Var START_MENU
+  Var DO_NOT_ADD_TO_PATH
+  Var ADD_TO_PATH_ALL_USERS
+  Var ADD_TO_PATH_CURRENT_USER
+  Var INSTALL_DESKTOP
+  Var IS_DEFAULT_INSTALLDIR
+;--------------------------------
+;Include Modern UI
+
+  Unicode True
```
Comment (nullkiller) [inline CI/NSIS.template.in:25]: Had to copy this from local NSIS installation just to add this line.

**Review by ArseniyShestakov — APPROVED** [review]

**Pull request merged**